homeassistant/components/wled/light.py

```diff
@@ -214,10 +219,29 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
         }
 
     @property
-    def hs_color(self) -> tuple[float, float]:
-        """Return the hue and saturation color value [float, float]."""
-        color = self.coordinator.data.state.segments[self._segment].color_primary
-        return color_util.color_RGB_to_hs(*color[:3])
+    def color_mode(self) -> str:
+        """Return the color mode of a light."""
+        if self._rgbw and self._wv:
+            return COLOR_MODE_RGBW
+        return COLOR_MODE_RGB
+
+    @property
+    def rgb_color(self) -> tuple[int, int, int] | None:
+        """Return the color value."""
+        if self._rgbw and self._wv:
+            return None
+        return self.coordinator.data.state.segments[self._segment].color_primary[:3]
+
+    @property
+    def rgbw_color(self) -> tuple[int, int, int, int] | None:
+        """Return the color value."""
+        if not self._rgbw or not self._wv:
+            return None
+
```

Comment:
There's no need to guard, the base entity will filter the attributes based on color_mode
```suggestion
```